static/js/src/advantage/set-modal-info.js

```diff
@@ -189,8 +189,14 @@ export function getPaymentInformation(paymentMethod) {
 }
 
 export function getRenewalInformation(data) {
-  const startDate = new Date(data.renewalStart);
-  const endDate = new Date(data.renewalEnd);
+  const current = new Date();
+  let startDate = current;
+
+  if (current > startDate) {
+    startDate = new Date(data.newContractStart);
+  }
```

Comment:
```suggestion
  const startDate = current > startDate ? new Date(data.newContractStart) : new Date();
```
[Ternary](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Conditional_Operator) baby!